engine/src/test/scala/cromwell/SimpleWorkflowActorSpec.scala

```diff
@@ -1,80 +1,144 @@
+    val workflowActor = TestFSMRef(new WorkflowActor(workflowId, StartNewWorkflow, workflowSources, ConfigFactory.load(),
+      watchActor,
+      system.actorOf(Props.empty, s"workflow-copy-log-router-$workflowId-${UUID.randomUUID()}")),
+      name = s"workflow-actor-$workflowId")
+    WorkflowActorandMetadataPromise(workflowActor, promise)
+  }
 
   val TestExecutionTimeout = 10.seconds.dilated
 
   "A WorkflowActor" should {
-    "start, run, succeed and die" ignore {
-//      startingCallsFilter("hello.hello") {
-//        val fsm = buildWorkflowFSMRef(SampleWdl.HelloWorld, SampleWdl.HelloWorld.wdlJson)
-//        val probe = TestProbe()
-//        probe watch fsm
-//        within(TestExecutionTimeout) {
-//          waitForPattern("transitioning from Submitted to Running") {
-//            waitForPattern("transitioning from Running to Succeeded") {
-//              fsm ! Start()
-//            }
-//          }
-//        }
-//        probe.expectTerminated(fsm, 10.seconds.dilated)
-//      }
+    "start, run, succeed and die" in {
+      startingCallsFilter("hello.hello") {
+        val WorkflowActorandMetadataPromise(workflowActor, _) = buildWorkflowActor(SampleWdl.HelloWorld, SampleWdl.HelloWorld.wdlJson)
+        val probe = TestProbe()
+        probe watch workflowActor
+        within(TestExecutionTimeout) {
+          waitForPattern("transition from ReadyToMaterializeState to MaterializationSuccessfulState") {
+            waitForPattern("transition from FinalizingWorkflowState to WorkflowSucceededState") {
+              workflowActor ! StartWorkflowCommand
+            }
+          }
+        }
+        probe.expectTerminated(workflowActor, 10.seconds.dilated)
+      }
     }
 
-    "fail to construct with missing inputs" ignore {
-//      intercept[IllegalArgumentException] {
-//        buildWorkflowFSMRef(SampleWdl.HelloWorld, rawInputsOverride = "{}")
-//      }
+    "fail to construct with missing inputs" in {
+      val workflowId = WorkflowId.randomId()
+      val failureMatcher = FailureMatcher("Required workflow input 'hello.hello.addressee' not specified.")
+      val WorkflowActorandMetadataPromise(workflowActor, promise) = buildWorkflowActor(SampleWdl.HelloWorld, "{}", workflowId, Option(failureMatcher))
+      val probe = TestProbe()
+      probe watch workflowActor
+      within(TestExecutionTimeout) {
+        waitForPattern("transition from MaterializingWorkflowDescriptorState to WorkflowFailedState") {
+          workflowActor ! StartWorkflowCommand
+          Await.result(promise.future, Duration.Inf)
+        }
+      }
+      probe.expectTerminated(workflowActor, 10.seconds.dilated)
     }
 
-    "fail to construct with inputs of the wrong type" ignore {
-//      intercept[IllegalArgumentException] {
-//        buildWorkflowFSMRef(SampleWdl.HelloWorld, rawInputsOverride = s""" { "$Addressee" : 3} """)
-//      }
+    "fail to construct with inputs of the wrong type" in {
```

Comment:
I'm not sure whether it is already... but this should be a `MaterializeWorkflowDescriptorActorSpec` test